logstash-core-event/lib/logstash/event.rb

```diff
@@ -92,13 +92,8 @@ def cancelled?
 
   # Create a deep-ish copy of this event.
   def clone
-    copy = {}
-    @data.each do |k,v|
-      # TODO(sissel): Recurse if this is a hash/array?
-      copy[k] = begin v.clone rescue v end
-    end
-
-    self.class.new(copy)
+    copy = LogStash::Util.deep_clone(@data)
+    self.class.new(copy.merge({ "@metadata" => LogStash::Util.deep_clone(@metadata.clone) }))
```

Comment:
I believe the `@metadata.clone` is unnecessary since deep_clone will take care of all the values.